src/figuretype/supplier.c

```diff
@@ -168,6 +168,27 @@ static int change_market_supplier_destination(figure *f, int dst_building_id)
     return 1;
 }
 
+static int new_market_dest_better_than_old_market_dest(figure *f, resource_type r, resource_storage_info info[RESOURCE_MAX])
+{
+    building *old_dest = building_get(f->destination_building_id);
+    // if any of these are true, the new building is automatically better
+    if (!building_is_active(old_dest)) {
+        return 1;
+    } else if (old_dest->type == BUILDING_GRANARY && old_dest->resources[r] <= 0) {
+        return 1;
+    } else if (old_dest->type == BUILDING_WAREHOUSE && building_warehouse_get_amount(old_dest, r) <= 0) {
+        return 1;
+    }
+    // make sure the new building is less than or equal to half the distance from the old
+    // building to help prevent market ladies from "ping ponging" back and forth
+    int old_dest_dist = building_dist(f->x, f->y, 1, 1, old_dest);
+    int half_old_dest_dist = old_dest_dist > 0 ? old_dest_dist / 2 : 0;
```

Comment:
```suggestion
    int half_old_dest_dist = old_dest_dist / 2;
```

No need for the condition, a division from `0` is safe.

In fact you don't even need the variable, just place `old_dest_dist / 2` on the `if` itself.

Reply:
Oh wow I'm dumb.